adapter/service/colorctrl/routing.go

```diff
@@ -0,0 +1,185 @@
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			colorctrl, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			color, err := message.Payload.GetIntMapValue()
+			if err != nil {
+				return nil, fmt.Errorf("adapter: failed to parse color: %w", err)
+			}
+
+			err = colorctrl.SetColor(color)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: failed to set color: %w", err)
+			}
+
+			_, err = colorctrl.SendColorReport(true)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: failed to send color report: %w", err)
+			}
+
+			return nil, nil
+		}),
+	)
+}
+
+// RouteCmdColorGetReport returns a routing responsible for handling the command.
+func RouteCmdColorGetReport(adapter adapter.Adapter) *router.Routing {
+	return router.NewRouting(
+		HandleCmdColorGetReport(adapter),
+		router.ForService(Colorctrl),
+		router.ForType(CmdColorGetReport),
+	)
+}
+
+// HandleCmdColorGetReport returns a handler responsible for handling the command.
+func HandleCmdColorGetReport(adapter adapter.Adapter) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			s := adapter.ServiceByTopic(message.Topic)
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			colorctrl, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			_, err := colorctrl.SendColorReport(false)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: failed to send color report: %w", err)
+			}
+
+			return nil, nil
+		}),
+	)
+}
+
+// RouteCmdColorStartTransition returns a routing responsible for handling the command.
+func RouteCmdColorStartTransition(adapter adapter.Adapter) *router.Routing {
+	return router.NewRouting(
+		HandleCmdColorStartTransition(adapter),
+		router.ForService(Colorctrl),
+		router.ForType(CmdColorStartTransition),
+	)
+}
+
+// HandleCmdColorStartTransition returns a handler responsible for handling the command.
+func HandleCmdColorStartTransition(adapter adapter.Adapter) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			s := adapter.ServiceByTopic(message.Topic)
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			colorctrl, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			var transitionObject map[string]interface{}
```

Comment:
The object has it's structure already defined in FIMP specification (https://github.com/futurehomeno/fimp-api#color-control-service), so it should be a struct rather than a map.